app/javascript/lib/validation/expression-matrices-validation.js

```diff
@@ -138,7 +253,7 @@ function validateColumnNumber(line, isLastLine, headers, lineNum, dataObj) {
 
     const msg = `All rows must have the same number of columns. ` +
       `Please ensure the number of columns for ${rowText}: ${notedBadRows}, ` +
-      `matches the file-header-specified number of ${correctNumberOfColumns} columns-per-row.`
+      `${containText} the same number of columns-per-row.`
```

Comment:
Compound modifiers are hyphenated if they come before the noun they modify (e.g. "columns-per-row number"), but not after (like here).

```suggestion
      `${containText} the same number of columns per row.`
```